sources/imperative/reader/Reader.swift

```diff
@@ -52,13 +52,22 @@ public final class CSVReader: IteratorProtocol, Sequence {
       guard !headers.isEmpty else { throw Error._invalidEmptyHeader() }
       self.headers = headers
       self.count = (rows: 1, fields: headers.count)
+    case let .lineNumber(index):
+        // Parse rows, but ignore them
+        for _ in 0..<index {
+            _ = try self._parseLine(rowIndex: 0)
+        }
+        guard let headers = try self._parseLine(rowIndex: index) else { self.status = .finished; return }
```

Comment:
By a quirk of how the public `rowIndex` is calculated this line should be `self._parseLine(rowIndex: 0)`.